docs/_posts/0000-01-03-continuous-delivery-and-deployment.md

```diff
@@ -253,27 +273,196 @@ Delete the branch… done.
+> Each line of development can be separated into a separate “topic branch” and maintained independently.
+> Branches in git are cheap, there is no reason to not make free use of them.
+
+[A Guide to the Kernel Development Process](https://www.kernel.org/doc/html/latest/process/7.AdvancedTopics.html) 
+
+--
+
+> ...Git is “expressly designed to make you feel less intelligent than you thought you were.” What’s your response to this?
+
+> Torvalds: So I think it used to be true but isn’t any more. There is a few reasons people feel that way, but I think only one of them remains. The one that remains is fairly simple: “you can do things so many ways.”
+
+[10 Years of Git](https://www.linuxfoundation.org/blog/2015/04/10-years-of-git-an-interview-with-git-creator-linus-torvalds)
+
+--
+
 ## A prophecy...
 
 <iframe src="https://player.vimeo.com/video/60788996" width="640" height="360" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
 <p><a href="https://vimeo.com/60788996">GIT IS HARD</a> from <a href="https://vimeo.com/owltastic">Meagan Fisher</a> on <a href="https://vimeo.com">Vimeo</a>.</p>
 
 Note:
+This video probably needs a bit of context setting - it's a developer having some issues with git which perplexes the whole team.
+This situation is not uncommon ;-)
 We're going to do some exercises involving branching, git is a very powerful tool, it's possible to get things VERY wrong.
 
 --
 
+<img style="border:0;box-shadow:none;background-color:transparent" src="images/aliens.png">
+
+Note:
+I'm increasingly of the opinion that there is only one real explanation for how git works, and that is that it is alien technology ;-)
+
+--
+
 <img width="500px" src="images/guitar_hero.png">
 
 Note:
 Branches can easily get out of control, they should be few in number and short-lived - you have been warned.
 
 --
 
-<img style="border:0;box-shadow:none;background-color:transparent" src="images/aliens.png">
+### Git guidelines
+
++ Use a small set of commands:
+    + clone
+    + commit
+    + push
+    + pull
++ Use a consistent workflow
++ Branches should be short-lived
++ The number of branches should be small
+
+--
+
+## Merging
+
+<img src="https://wac-cdn.atlassian.com/dam/jcr:09308632-38a3-4637-bba2-af2110629d56/07.svg?cdnVersion=ks">
+
 
 Note:
-I'm increasingly of the opinion that there is only one real explanation for how git works, and that is that it is alien technology ;-)
+What is a merge
+Sometimes it's easy
+And sometimes it's hard
+
+--
+
+<backgroundimage>images/traffic_merge.svg</backgroundimage>
+<backgroundimageopacity>0.25</backgroundimageopacity>
+
++ Git is actually good at merging
++ Some merges are simple
++ Some merges are not
++ Keeping your branches short-lived decreases the risk of complicated merges
+    
+
+--
+
+
+### Different kinds of merge
+
+* Simple
+    + Different files
+    + Same file, different lines
+    + Same file, same lines
+* Complicated
```

Comment:
I think these should be fragments… switch the `*`s for `+`s?

Reply:
Actually they are intentionally bullets, supposed to represent a scale from simple to complicated - let me know if you think it's too confusing.